src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -452,58 +280,37 @@ def to_pex_request(
             sources=sources,
         )
 
-    @staticmethod
-    async def _find_python_interpreter_constraints_from_lockfile(
-        subsystem: PythonToolBase,
-    ) -> InterpreterConstraints:
-        """If a lockfile is used, will try to find the interpreter constraints used to generate the
-        lock.
 
-        This allows us to work around https://github.com/pantsbuild/pants/issues/14912.
-        """
-        # If the tool's interpreter constraints are explicitly set, or it is not using a lockfile at
-        # all, then we should use the tool's interpreter constraints option.
-        if (
-            not subsystem.options.is_default("interpreter_constraints")
-            or not subsystem.uses_lockfile
-        ):
-            return subsystem.interpreter_constraints
-
-        # If using Pants's default lockfile, we can simply use the tool's default interpreter
-        # constraints, which we trust were used to generate Pants's default tool lockfile.
-        if not subsystem.uses_custom_lockfile:
-            return InterpreterConstraints(subsystem.default_interpreter_constraints)
-
-        # Else, try to load the metadata block from the lockfile.
-        requirements = subsystem.pex_requirements()
-        assert isinstance(requirements, EntireLockfile)
-        lockfile = await Get(LoadedLockfile, LoadedLockfileRequest(requirements.lockfile))
-        return (
-            lockfile.metadata.valid_for_interpreter_constraints
-            if lockfile.metadata
-            else subsystem.interpreter_constraints
-        )
+async def get_loaded_lockfile(subsystem: PythonToolBase) -> LoadedLockfile:
```

Comment:
Some useful rule helpers, that made sense to keep here.

Reply:
To confirm, these aren't new right? I didn't closely review them.

Reply:
They are new-ish actually. They replace the old `_find_python_interpreter_constraints_from_lockfile`, and improve on it.